pkg/dyninst/actuator/actuator.go

```diff
@@ -241,11 +274,11 @@ func (a *effects) attachToProcess(
 var detachLogLimiter = rate.NewLimiter(rate.Every(1*time.Minute), 10)
 
 // detachFromProcess detaches a program from a process.
-func (a *effects) detachFromProcess(ap *attachedProgram) {
+func (a *effects) detachFromProcess(ap *attachedProgram, failure error) {
 	a.wg.Add(1)
 	go func() {
 		defer a.wg.Done()
-		if err := ap.attachedProgram.Detach(); err != nil {
+		if err := ap.attachedProgram.Detach(failure); err != nil {
```

Comment:
I feel like we want to log somewhere (maybe in the caller of this method, maybe here, or maybe in Detach) when we hit these issues. Right now there'd be no evidence. I could also see incrementing a counter. Given the possibility that incrementing a counter is the right thing, perhaps the place to do the logging and incrementing is underneath the Detach call in `module`. Perhaps a structured error might be useful (or not).